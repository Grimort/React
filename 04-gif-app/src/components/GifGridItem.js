import React from "react";

const GifGridItem = ({ id, title, url }) => {
  return (
    <div className="card animate__animated animate__bounce animate__fadeInDown">
      <p>{title}</p>
      <img src={url.url} alt={title}></img>
    </div>
  );
};

export default GifGridItem;
