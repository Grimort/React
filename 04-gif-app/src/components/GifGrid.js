import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn" key={category}>
        {category}
      </h3>
      <div className="card-grid animate__animated animate__bounce animate__fadeIn">
        {loading && (
          <p className="animate__animated animate__flash">Loading...</p>
        )}
        {images.map((img) => (
          <GifGridItem key={img.id} url={img.url.url} title={img.title} />
        ))}
      </div>
    </>
  );
};
GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
