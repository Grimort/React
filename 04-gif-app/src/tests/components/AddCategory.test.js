import React from "react";
import "@testing-library/jest-dom/matchers";
import "@testing-library/jest-dom/extend-expect";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("Testing AddCategory component", () => {
  const setCategories = jest.fn();
  let wrapper;
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test("should to show the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should change the input value", () => {
    const input = wrapper.find("input");
    const value = "Memes";
    input.simulate("change", { target: { value } });
    expect(wrapper.find("p").text().trim()).toBe(value);
  });
  test("shouldn't call the function setCategories", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("should call setCategories and input field empty", () => {
    const input = wrapper.find("input");
    const value = "Simpsons";
    input.simulate("change", { target: { value } });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(input.prop("value")).toBe("");
  });
});
