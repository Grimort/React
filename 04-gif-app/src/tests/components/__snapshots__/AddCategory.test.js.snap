// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory component should to show the component correctly 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <p />
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
